src/spreadsheet/tokenizer.py

```diff
@@ -0,0 +1,165 @@
+import io
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from tokenize import ENDMARKER, NAME, NEWLINE, TokenError, generate_tokens
+from tokenize import NUMBER as PY_NUMBER
+from tokenize import STRING as PY_STRING
+from typing import List
+
+POSTFIX_UNARY_OPERATORS = ["%"]
+OPERATORS = "+,-,*,/,=,<>,>=,>,<=,<,^,&".split(",") + POSTFIX_UNARY_OPERATORS
+
+
+NUMBER = "NUMBER"
+STRING = "STRING"
+BOOLEAN = "BOOLEAN"
+UNKNOWN = "UNKNOWN"
+OPERATOR = "OPERATOR"
+SYMBOL = "SYMBOL"
+FUNCTION = "FUNCTION"
+LEFT_PAREN = "LEFT_PAREN"
+RIGHT_PAREN = "RIGHT_PAREN"
+COMMA = "COMMA"
+DEBUGGER = "DEBUGGER"
```

Comment:
Define these as [an Enum](https://docs.python.org/3/library/enum.html)?

Reply:
this version of the tokenizer didn't work. I converted the complete one of o-spreadsheet.